src/java/com/puppetlabs/puppetserver/ExecutionStubImpl.java

```diff
@@ -1,8 +1,12 @@
 package com.puppetlabs.puppetserver;
 
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.apache.commons.exec.CommandLine;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 
 public class ExecutionStubImpl {
```

Comment:
As this class is no longer a Stub (it will be used in more than just the one jruby puppet method call), we should probably rename it. Any suggestions? Here are a few ideas: ExecutionShell, ShellExecutor, ShellCommons...